src/Workspaces/Core/Portable/LanguageServices/TypeInferenceService/AbstractTypeInferenceService.cs

```diff
@@ -73,6 +73,16 @@ private IEnumerable<ITypeSymbol> Filter(IEnumerable<ITypeSymbol> types, bool fil
                             .ToImmutableReadOnlyListOrEmpty();
             }
 
+            protected IEnumerable<ITypeSymbol> ExpandParamsParameter(IParameterSymbol p)
+            {
+                var result = SpecializedCollections.SingletonEnumerable(p.Type);
+                if (p.IsParams)
+                {
+                    result = result.Concat((p.Type as IArrayTypeSymbol).ElementType);
```

Comment:
Rather than the as, do a hard cast so we get a better diagnostic error if we have a params that isn't IArrayTypeSymbol?